sdk/compute/Azure.ResourceManager.Compute/src/Customize/SshPublicKeyResource.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Compute.Models;
+using Azure.ResourceManager.Resources;
+
+namespace Azure.ResourceManager.Compute
+{
+    public partial class SshPublicKeyResource
+    {
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual async Task<Response<SshPublicKeyGenerateKeyPairResult>> GenerateKeyPairAsync(CancellationToken cancellationToken = default)
+        {
+            using var scope = _sshPublicKeyClientDiagnostics.CreateScope("SshPublicKeyResource.GenerateKeyPair");
+            scope.Start();
+            try
+            {
+                var response = await _sshPublicKeyRestClient.GenerateKeyPairAsync(Id.SubscriptionId, Id.ResourceGroupName, Id.Name, cancellationToken).ConfigureAwait(false);
+                return response;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
```

Comment:
Please change the implementation here, to just redirect the call to the new `GenerateKeyPairAsync` but with a null body.
```suggestion
        public virtual async Task<Response<SshPublicKeyGenerateKeyPairResult>> GenerateKeyPairAsync(CancellationToken cancellationToken = default)
        {
            return await GenerateKeyPairAsync(null, cancellationToken).ConfigureAwait(false);
        }
```

Reply:
my worry is with this approach, anytime there is more parameters added to the function, we will also need to update this customization, and adding to maintenance cost. Please let me know your thoughts

Reply:
@ArcturusZhang

Reply:
it is inevitable.
Your previous approach has even more maintenance cost. We could only reduce the cost.

The generator could help in scenarios like this to generate those overloads automatically for you. We have this feature in plan, but not fully implemented yet. Our first step for this feature is to support generating overloads for the model factory because breaking changes there happen much more frequently

Reply:
another way to look at its maintenance cost here is that: in the way I suggested, we are calling a **public** method. We need to keep all the public method invocation work as the same when there is an update.
Therefore if everything is working properly, the code you wrote here, should **never** need a follow up change even if this method receives another parameter, because this method is calling the **public** method and it has to exist in every version of the library.

Let me summary, using this simple case:
in v1 we have
```
public virtual Response Foo(CancellationToken token = default);
```
in v2 we have
```
public virtual Response Foo(CancellationToken token) => Foo(null, token); // this is your manual code
public virtual Response Foo(string v2 = null, CancellationToken token = default);
```
in v3 we have
```
public virtual Response Foo(CancellationToken token) => Foo(null, token); // this does not need a change
public virtual Response Foo(string v2, CancellationToken token = default) => Foo(v2, null, token); // this is your new manual code
public virtual Response Foo(string v2 = null string v3 = null, CancellationToken token = default);
```

The point here is that our approaches are basically the same, every time you introduce a new parameter, you HAVE TO add a bunch of code, this is the inevitable cost.
But my point is that we could reduce the amount of that, and we should never need to change those have been written in the previous version.

Reply:
thank you. addressed  !